includes/wsu-content-type-announcement.php

```diff
@@ -125,6 +127,42 @@ public function enqueue_admin_scripts() {
 	}
 
 	/**
+	 * Display the meta box used to capture an external URL for
+	 * an announcement.
+	 *
+	 * @since 1.9.0
+	 *
+	 * @param \WP_Post $post
+	 */
+	public function display_url_meta_box( $post ) {
+		$url = get_post_meta( $post->ID, '_announcement_url', true );
+
+		if ( $url ) {
+			$url_parts = wp_parse_url( $url );
+
+			if ( false === $url_parts ) {
+				$url = '';
+			} elseif ( ! isset( $url_parts['scheme'] ) ) {
+				$url = esc_url( 'http://' . $url_parts['host'] . $url_parts['path'] );
+			} else {
+				$url = esc_url( $url_parts['scheme'] . '://' . $url_parts['host'] . $url_parts['path'] );
+			}
+		} else {
+			$url = '';
+		}
+
+		?>
+		<p>Enter the full URL of an announcement here if the posted elsewhere.</p>
+		<label for="announcement-form-url">URL:</label>
+		<input type="text"
+				id="announcement-form-url"
+				name="announcement_url"
+				style="width: 575px; max-width: 100%;"
```

Comment:
Without having tried it, `width:100%;` on its own might work here just as well.

Reply:
The initial with of 575 makes the input appear to the right of the label until the screen is small enough, then it drops below. Not entirely necessary, but lazy formatting :)

Reply:
👍